cmd/odf/restore/crds.go

```diff
@@ -20,8 +63,54 @@ var deletedCmd = &cobra.Command{
 	},
 	Run: func(cmd *cobra.Command, args []string) {
 		k8sutil.SetDeploymentScale(cmd.Context(), root.ClientSets.Kube, root.OperatorNamespace, "ocs-operator", 0)
-		pkgrestore.RestoreCrd(cmd.Context(), root.ClientSets, root.OperatorNamespace, root.StorageClusterNamespace, args)
-		k8sutil.SetDeploymentScale(cmd.Context(), root.ClientSets.Kube, root.OperatorNamespace, "ocs-operator", 1)
+		// Parse the fully qualified CRD (e.g. "cephclusters.ceph.rook.io").
+		resourceName, groupName, version, err := parseFullyQualifiedCRD(args[0])
+		if err != nil {
+			logging.Fatal(fmt.Errorf("Error parsing CRD type: %v\n", err))
+		}
+		// Construct a new args slice with the resource name as the first argument.
+		newArgs := make([]string, len(args))
+		newArgs[0] = resourceName
+		if len(args) > 1 {
+			newArgs[1] = args[1]
+		}
+		var customResources []pkgrestore.CustomResource
+		if slices.Contains(newArgs, "storageclusters") {
+			customResources = []pkgrestore.CustomResource{
+				// ceph.rook.io/v1
+				{Group: "ceph.rook.io", Version: "v1", Resource: "cephblockpoolradosnamespaces"},
```

Comment:
how can we create a link back to the ocs-op to keep this list updated when a new resource is owned by storagecluster?

did you check that k8s primitives (secrets, configmaps etc) that are owned by storagecluster are ok to be deleted?

apart from ensuring pods are running what other checks are performed after restoration of storagecluster CR?

iirc, storagecluster UID is being used in DR related activities as well, maybe that also gets effected or we need to have a statement of not performing this on DR enabled setups?

Reply:
@leelavg I have not tested it on a DR setup or in Provider mode.